packages/starlight/schemas/badge.ts

```diff
@@ -1,10 +1,14 @@
 import { z } from 'astro/zod';
 
-const badgeSchema = () =>
-	z.object({
-		variant: z.enum(['note', 'danger', 'success', 'caution', 'tip', 'default']).default('default'),
-		text: z.string(),
-	});
+const badgeSchema = () => z.object({
+	variant: z.enum(['note', 'danger', 'success', 'caution', 'tip', 'default']).default('default'),
+	text: z.string(),
+	class: z.string().optional(),
+});
+
+export const BadgeComponentSchema = badgeSchema().extend({
+	size: z.enum(['small', 'medium', 'large']).default('small'),
+});
```

Comment:
```suggestion
const badgeSchema = () =>
	z.object({
		variant: z.enum(['note', 'danger', 'success', 'caution', 'tip', 'default']).default('default'),
		text: z.string(),
		class: z.string().optional(),
	});

export const BadgeComponentSchema = badgeSchema()
	.extend({
		size: z.enum(['small', 'medium', 'large']).default('small'),
	})
	.passthrough();

export type BadgeComponentProps = z.input<typeof BadgeComponentSchema>;
```

We can use [`.passthrough`](https://github.com/colinhacks/zod#passthrough) so that unknown keys are not stripped.